File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-1.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    private boolean isSubRoot = false;  
    public boolean isSubtree(TreeNode root, TreeNode subRoot) {

        StringBuilder navigationPathRoot = new StringBuilder();
        StringBuilder navigationPathSubRoot = new StringBuilder();
        navigate(subRoot, navigationPathSubRoot);
        navigateStartingFromCurrentNode(root, navigationPathSubRoot, subRoot);
        return isSubRoot;
    }

    private void navigate(TreeNode node, StringBuilder sb){
        sb.append(node.val);
        if(node.left != null){
            sb.append("left");
            navigate(node.left, sb);
        }
        if(node.right != null){
            sb.append("right");
            navigate(node.right, sb); 
        }   
    }

    private void navigateStartingFromCurrentNode(TreeNode currentNode, StringBuilder sb, TreeNode subTree){
        if(currentNode == null)
            return;

        StringBuilder sbNew = new StringBuilder();
        navigate(currentNode, sbNew);
        if(sbNew.toString().equals(sb.toString())){
            this.isSubRoot = true;
            return;
        }
        navigateStartingFromCurrentNode(currentNode.left, sb, subTree);
        navigateStartingFromCurrentNode(currentNode.right, sb, subTree);

    }
}

File: Data Structures & Algorithms/same-binary-tree/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isSameTree(self, p: Optional[TreeNode], q: Optional[TreeNode]) -> bool:
        navigationPathP = [];
        navigationPathP.append("root");
        navigationPathQ = [];
        navigationPathQ.append("root");
        self.navigate(p, navigationPathP);
        self.navigate(q, navigationPathQ);
        return navigationPathP == navigationPathQ;

    def navigate(self, node: Optional[TreeNode], navigationPath: list[str]) -> None:
        if(node is None):
            return;
        navigationPath.append(node.val)
        navigationPath.append("left")
        self.navigate(node.left, navigationPath)
        navigationPath.append("right")
        self.navigate(node.right, navigationPath)       